c/planarityTestGraphFunctionality.c

```diff
@@ -408,15 +315,120 @@ int testAllGraphs(graphP theGraph, char command, char *inputStr, testAllStatsP s
 	if (freeG6ReadIterator(&pG6ReadIterator) != OK)
 		ErrorMessage("Unable to free G6ReadIterator.\n");
 	
-	return exitCode;
+	return Result;
 }
 
-int _getNumCharsToReprInt(int theNum)
+int outputTestAllGraphsResults(char command, testAllStatsP stats, char * infileName, char *outfileName, char **outputStr)
 {
-	int numCharsRequired = 1;
+	int Result = OK;
 
-	while(theNum /= 10)
-		numCharsRequired++;
+	FILE *outputFileP;
 
-	return numCharsRequired;
-}
\ No newline at end of file
+	int charsAvailForFilename = 0;
+	char *messageFormat = NULL;
+	char messageContents[MAXLINE + 1];
+
+	strOrFileP testOutput = NULL;
+
+	if (outfileName != NULL)
+	{
+		outputFileP = fopen(outfileName, "w");
+		if (outputFileP == NULL)
+		{
+			charsAvailForFilename = (int) (MAXLINE - strlen(outfileName));
+			messageFormat = "Unable to open file \"%.*s\" for output.\n";
+			sprintf(messageContents, messageFormat, charsAvailForFilename, outfileName);
+			ErrorMessage(messageContents);
+			return NOTOK;
+		}
+		
+		testOutput = sf_New(outputFileP, NULL);
+	}
+	else
+	{
+		if ((*outputStr) == NULL)
+			(*outputStr) = (char *) malloc(1 * sizeof(char));
+		
+		if ((*outputStr) == NULL)
+		{
+			ErrorMessage("Unable to allocate memory for outputStr.\n");
+			return NOTOK;
+		}
+
+		testOutput = sf_New(NULL, (*outputStr));
+	}
+
+	if (testOutput == NULL)
+	{
+		ErrorMessage("Unable to set up string-or-file container for test output.\n");
+
+		if (outputFileP != NULL)
+		{
+			fclose(outputFileP);
+			outputFileP = NULL;
+		}
+
+		if ((*outputStr) != NULL)
```

Comment:
This seems like it would have the potential to crash because the double pointer outputStr could be NULL if an output file name was given. That being said, if you free the output string above (if the second sf_New fails), then you won't have this if test because you'll be in the right branch of code.